src/query/functions/temporal/base.go

```diff
@@ -266,12 +262,14 @@ func parallelProcess(
 			series     = iter.Current()
 			datapoints = series.Datapoints()
 			stats      = series.Stats()
+			seriesMeta = metas[i]
```

Comment:
paranoia: Worth checking if within bounds at all?

Reply:
Yeah, may as well just in case :+1: